bluemira/geometry/parameterisations.py

```diff
@@ -986,6 +986,191 @@ def _make_control_points(p0, p3, theta0, theta3, l_start, l_end):
+        axis,
+        x_out,
+        x_curve_start,
+        x_mid,
+        z_top,
+        z_mid,
+        flip=False,
+        r_c=0,
+    ):
+        """
+        Makes smooth dome for CP coils
+
+        Returns
+        -------
+        shape: BluemiraWire
+            CAD Wire of the geometry
+
+        """
+        # Define basic Top Curve (with no joint or corner transitions)
+
+        out_wire = []
+
+        # Define the basic main curve (with no joint or transitions curves)
+        alpha = np.arctan(0.5 * (x_out - x_curve_start) / abs(z_top - z_mid))
+        theta_leg_basic = 2 * (np.pi - 2 * alpha)
+        r_leg = 0.5 * (x_out - x_curve_start) / np.sin(theta_leg_basic / 2)
+        z_top_r_leg = z_top + r_leg if flip else z_top - r_leg
+        leg_centre = (x_out - 0.5 * (x_out - x_curve_start), 0, z_top_r_leg)
+
+        # Transitioning Curve
+        sin_a = np.sin(theta_leg_basic / 2)
+        cos_a = np.cos(theta_leg_basic / 2)
+        alpha_leg = (
+            np.arcsin(np.abs(r_leg * sin_a - r_c) / (r_leg - r_c)) + theta_leg_basic / 2
+        )
+
+        # Joint Curve
+        r_j = min(x_curve_start - x_mid, 0.8)
+        theta_j = np.arccos((r_leg * cos_a + r_j) / (r_leg + r_j))
+        z_mid_r_j = z_mid - r_j if flip else z_mid + r_j
+        joint_curve_centre = (
+            leg_centre[0] - (r_leg + r_j) * np.sin(theta_j),
+            0,
+            z_mid_r_j,
+        )
+
+        # Final total angular extent of main curve
+        theta_leg_final = alpha_leg - (theta_leg_basic / 2 - theta_j)
+
+        # Build straight section of leg
+        p1 = [x_mid, 0, z_mid]
+        p2 = [leg_centre[0] - (r_leg + r_j) * np.sin(theta_j), 0, z_mid]
+
+        if not flip:
+            out_wire.append(make_polygon([p1, p2]))
+        else:
+            out_wire.append(make_polygon([p2, p1]))
+        out_wire.append(
+            make_circle(
+                radius=r_j,
+                center=joint_curve_centre,
+                start_angle=90 - np.rad2deg(theta_j) if flip else -90,
+                end_angle=90 if flip else np.rad2deg(theta_j) - 90,
+                axis=axis,
+                label="bottom_limb_joint" if flip else "top_limb_joint",
+            )
+        )
+
+        angle2 = np.rad2deg(theta_leg_final)
+        start2 = 90 + np.rad2deg(theta_j)
+        out_wire.append(
+            make_circle(
+                radius=r_leg,
+                center=leg_centre,
+                start_angle=-start2 if flip else start2 - angle2,
+                end_angle=angle2 - start2 if flip else start2,
+                axis=axis,
+                label="bottom_limb_dome" if flip else "top_limb_dome",
+            )
+        )
+
+        label = "bot_limb_dome" if flip else "top_limb_dome"
+
+        return BluemiraWire(out_wire, label=label)
+
+    @staticmethod
+    def _make_flat_leg(axis, x_in, x_out, z, r_i, r_o, flip=False):
+        """
+        Makes a flat leg with the option of one end rounded
```

Comment:
Which part of the leg is flat, the top (horizontal) or the outboard (vertical) ?

Also specify the arguments